x-pack/plugins/upgrade_assistant/__jest__/client_integration/overview/fix_issues_step/fix_issues_step.test.tsx

```diff
@@ -166,14 +168,18 @@ describe('Overview - Fix deprecation issues step', () => {
       const { exists, find } = testBed;
 
       expect(exists('kibanaStatsPanel')).toBe(true);
-      expect(find('kibanaStatsPanel.warningDeprecations').text()).toContain('1');
-      expect(find('kibanaStatsPanel.criticalDeprecations').text()).toContain('1');
+      expect(find('kibanaStatsPanel.warningDeprecations').text()).toContain(
+        mockedWarningKibanaDeprecations.length
+      );
+      expect(find('kibanaStatsPanel.criticalDeprecations').text()).toContain(
+        mockedCriticalKibanaDeprecations.length
+      );
     });
 
     test(`Hides deprecation count if it doesn't have any`, async () => {
       await act(async () => {
         const deprecationService = deprecationsServiceMock.createStartContract();
-        deprecationService.getAllDeprecations = jest.fn().mockRejectedValue([]);
+        kibanaDeprecationsServiceHelpers.setLoadDeprecations({ deprecationService, response: [] });
```

Comment:
It looks like the implementation of `setLoadDeprecations` will invoke `mockReturnValue`, but the original code invoked `mockRejectedValue`. If the implementation of the feature didn't change to resolve instead of reject, then why would this change to the test still work? 🤔

Reply:
I think the previous code was a mistake, as the test is checking the UI when there are no deprecations (not if there is an error). I think it was falsely passing before becomes there was no `component.update()` after the testbed setup.